site/src/routes/metrics.js

```diff
@@ -0,0 +1,154 @@
+import { stores } from '../constants.js'
+import { get as getDeals } from '../models/deals.js'
+
+const MAX_AGE_SECS = 60 * 10 // max age of a metrics response in seconds
+
+/**
+ * TODO: basic auth
+ * @param {FetchEvent} event
+ */
+export async function metrics(event) {
+  const cacheKey = new Request(new URL(event.request.url).toString(), event.request)
```

Comment:
Mind adding a code explaining when new Request is needed here as opposed to just using `event.request`. Reading this I have 0 guesses.

Reply:
Ya will do. I'm guessing since it's used as the cache key it's to remove varying client headers.

Reply:
since this is a pretty basic endpoint you can just use `/metrics` as the key no need to do `new Request` and `new URL().toString()` but i would add a cache busting string at the end to avoid issues when code changes and a new deploy is done. Check this https://github.com/filecoin-project/specs/blob/master/api/src/get.js for and example

Reply:
hmm, but `event.request` is passed in to the constructor so it'll inherit those headers...

It's based on this btw: https://developers.cloudflare.com/workers/examples/cache-api